### ISSUE TITLE
[Bug report] basic operation on DataSetList fails

### ISSUE THREAD

**Issue by nikohansen** [numbbo/coco-postprocess#44]
```python
import cocopp
dslist = cocopp.load('bbob/2009/B*')
dslist.dictByAlgName()
````
<img width="756" alt="Screen Shot 2022-05-27 at 21 46 11" src="https://user-images.githubusercontent.com/7316439/170780039-6d6b7b79-347e-4162-b398-0b76bab57ec9.png">

Now I can't really use the loaded data (I want to use `cocopp.compall.pprldmany.main` without going through `cocopp.main`)

**Comment by brockho** [comment]
Indeed, `dictByAlgName()` does not work, but `dictByAlg()` does---although with keys that are pairs of algorithm name and (empty) folder name such that

```python
dslist.dictByAlg().keys()
```
gives
```python
odict_keys([('BAYEDA_gallagher', ''), ('BFGS_ros', ''), ('BIPOP-CMA-ES_hansen', '')])
```
